designs/2023-rule-performance-dashboard/README.md

```diff
@@ -0,0 +1,210 @@
+- Repo: eslint/eslint
+- Start Date: 2023-03-10
+- RFC PR:
+- Authors: Mara Nikola Kiefer (@mnkiefer)
+
+# Rule Performance Dashboard
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+This document describes how to expose ESLint's [`TIMING`](https://eslint.org/docs/latest/extend/custom-rules#profile-rule-performance) information to the [formatters](https://eslint.org/docs/latest/use/formatters/) and use it in a new formatter [`html-charts`]() to generate a **Rule Performance Dashboard**.
+
+## Motivation
+
+<!-- Why are we doing this? What use cases does it support? What is the expected
+outcome? -->
+
+Analysing rule performance currently requires additional scripting to collect/extract more *granular* timing data (lint time per file per rule):
+
+1. Running ESLint per rule, per file and then collecting the file/rule time data output:
+
+  ```bash
+  TIMING=1 DEBUG=eslint:cli-engine eslint --no-eslintrc --rule ... file
+  ```
+
+2. Doing a single ESLint run on all files and then extracting the file/rule time data output:
+
+  ```bash
+  TIMING=all DEBUG=eslint:cli-engine eslint ... files
+  ```
+
+In addition, one needs to create an overview for an effective presentation of the results.
+
+It would be more *convenient and shareable* to have:
+
+   1. The **timing data** ESLint already collects exposed to the formatters
+   2. A **built-in formatter** for that data
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+The *proof-of-concept* can be found at:
+- [**ESLint**](https://github.com/mnkiefer/eslint/pull/1): **Fork of ESLint** with the above mentioned changes.
+- [**Sample project**](https://github.com/mnkiefer/eslint-samples): Example on which the dashboard (shown below) was generated.
+
+### ESLint *timing* exposed
+
+Exposure of the timing object requires only a few changes in the Linter and CLI engline.
+
+The timing is already collected in `lib/linter/timing.js`. Note, that the function requires the extra input paramter `filename`. For the function itself, the only change is that we now *persist* that information in the `lintTimesPerRule` object, which stores the detailed *per file per rule* lint times under the **lintTimes** key. An extra function `getLintTime()` is returned to the linter to retrieve this object.
```

Comment:
> An extra function `getLintTime()` is returned to the linter to retrieve this object.

Can you explain this further? It's not clear what "is returned to the linter" means.

Reply:
This is just a helper function in `timing.js`, similar to `getListSize()`. As it's just a technical detail, it should probably not be mentioned at all, so I will remove this sentence.